File: jvm-tools/XmlFormatter/src/test/kotlin/dev/vighnesh153/xml/formatter/fmt/FormatTextNodeTest.kt
package dev.vighnesh153.xml.formatter.fmt

import dev.vighnesh153.xml.formatter.ast.XmlTextNode
import dev.vighnesh153.xml.formatter.lexer.LexerInputReader
import dev.vighnesh153.xml.formatter.lexer.XmlLexer
import dev.vighnesh153.xml.formatter.parser.XmlParser
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class FormatTextNodeTest {
    private val lexerInputReader =
        LexerInputReader(input = "< pokemon  >  Pikachu    < /  pokemon >")
    private val lexer = XmlLexer(lexerInputReader)
    private val parser = XmlParser(lexer)
    private val program = parser.parseProgram()

    @Test
    fun formatsWithZeroIndentation() {
        assertEquals(0, parser.errors.size)
        assertEquals(1, program.statements.size)
        assertEquals(
            "Pikachu",
            (program.statements.first() as XmlTextNode).format(
                indentationLevel = 0,
                indentation = 4,
            )
        )
    }

    @Test
    fun formatsWithTwoIndentationLevel() {
        assertEquals(0, parser.errors.size)
        assertEquals(1, program.statements.size)
        assertEquals(
            "        Pikachu",
            (program.statements.first() as XmlTextNode).format(
                indentationLevel = 2,
                indentation = 4,
            )
        )
    }

    @Test
    fun formatsWithTwoIndentationLevelAndThreeIndentation() {
        assertEquals(0, parser.errors.size)
        assertEquals(1, program.statements.size)
        assertEquals(
            "      Pikachu",
            (program.statements.first() as XmlTextNode).format(
                indentationLevel = 2,
                indentation = 3,
            )
        )
    }
}

File: nodejs-tools/nodejs-compiler-tools/formatter-xml/src/format_text_node.test.ts
import { expect, test } from "vitest";
import { XmlTagNode, XmlTextNode } from "@vighnesh153/parser-xml";
import { formatTextNode } from "./format_text_node.ts";
import { parseProgram } from "./test_utils.ts";

test("should format text node with 0 indentation level", () => {
  const [parser, program] = parseProgram(
    `< pokemon  >  Pikachu    < /  pokemon >`,
  );

  expect(parser.errors.length).toBe(0);
  expect(program.statements.length).toBe(1);
  expect(
    formatTextNode({
      textNode: (program.statements[0] as XmlTagNode)
        .children[0] as XmlTextNode,
      indentation: 4,
      indentationLevel: 0,
    }),
  ).toMatchInlineSnapshot(`"Pikachu"`);
});

test("should format text node with 2 indentation level", () => {
  const [parser, program] = parseProgram(
    `< pokemon  >  Pikachu    < /  pokemon >`,
  );

  expect(parser.errors.length).toBe(0);
  expect(program.statements.length).toBe(1);
  expect(
    formatTextNode({
      textNode: (program.statements[0] as XmlTagNode)
        .children[0] as XmlTextNode,
      indentation: 4,
      indentationLevel: 2,
    }),
  ).toMatchInlineSnapshot(`"        Pikachu"`);
});

test("should format text node with 2 indentation level and 3 indentation", () => {
  const [parser, program] = parseProgram(
    `< pokemon  >  Pikachu    < /  pokemon >`,
  );

  expect(parser.errors.length).toBe(0);
  expect(program.statements.length).toBe(1);
  expect(
    formatTextNode({
      textNode: (program.statements[0] as XmlTagNode)
        .children[0] as XmlTextNode,
      indentation: 3,
      indentationLevel: 2,
    }),
  ).toMatchInlineSnapshot(`"      Pikachu"`);
});
